libp2p/stream/connection.nim

```diff
@@ -95,22 +100,13 @@ proc pollActivity(s: Connection): Future[bool] {.async.} =
   # Inactivity timeout happened, call timeout monitor
 
   trace "Connection timed out", s
-  if not(isNil(s.timeoutHandler)):
+  if s.timeoutHandler != nil:
     trace "Calling timeout handler", s
-
-    try:
-      await s.timeoutHandler()
-    except CancelledError:
-      # timeoutHandler is expected to be fast, but it's still possible that
-      # cancellation will happen here - no need to warn about it - we do want to
-      # stop the polling however
-      debug "Timeout handler cancelled", s
-    except CatchableError as exc: # Shouldn't happen
-      warn "exception in timeout handler", s, exc = exc.msg
+    await s.timeoutHandler()
```

Comment:
Why has the try been removed?

Reply:
`timeoutHandler` has `{.async: (raises: []).}` so is statically checked at compiletime to never raise. so, the exception handlers are unreachable.